events/src/utils.py

```diff
@@ -313,6 +314,56 @@ def plugin_webhook(message):
     webhooks_pool.send(message["event"], message["ts"], message_json)
 
 
+class NamedTempOpen(object):
+    """
+        This class is used to create a temporary file which is then written to with contents.
+        The temp file is then persisted with the give name by calling os.rename().
+        This class is used to avoid the data loss or truncation in case of multiple processes
+        writing to the same file without the use of fcntl locks.
+
+        The temporary file is created in the dest dir of the file.
+    """
+
+    def __init__(self, filename, open_mode, *args, **kwagrs) -> None:
+        self.filename = filename
+        self.open_mode = open_mode
+        self.open_args = args
+        self.open_kwargs = kwagrs
+        self.working_dir = "."
+        self.fileobj = None
+        self.file_persisted = False
+
+        if os.path.dirname(self.filename) != "":
+            self.working_dir = os.path.dirname(self.filename)
+
+    def persist(self):
+        os.rename(self.fileobj.name, self.filename)
+        self.file_persisted = True
```

Comment:
This is not needed. It can be done inside `__exit__()`. See comment in `__exit__()`.